crd/multitenantpodnetworkconfig/client.go

```diff
@@ -0,0 +1,161 @@
+	mpnc, err := GetMultitenantPodNetworkConfigs()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get embedded mpnc crd")
+	}
+	current, err := i.create(ctx, mpnc)
+	if !apierrors.IsAlreadyExists(err) {
+		return current, err
+	}
+	if current == nil {
+		current, err = i.cli.Get(ctx, mpnc.Name, metav1.GetOptions{})
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to get existing mpnc crd")
+		}
+	}
+	if !reflect.DeepEqual(mpnc.Spec.Versions, current.Spec.Versions) {
+		mpnc.SetResourceVersion(current.GetResourceVersion())
+		previous := *current
+		current, err = i.cli.Update(ctx, mpnc, metav1.UpdateOptions{})
+		if err != nil {
+			return &previous, errors.Wrap(err, "failed to update existing mpnc crd")
+		}
+	}
+	return current, nil
+}
+
+// Client provides methods to interact with instances of the MultitenantPodNetworkConfig custom resource.
+type Client struct {
+	cli client.Client
+}
+
+// NewClient creates a new MultitenantPodNetworkConfig client around the passed ctrlcli.Client.
+func NewClient(cli client.Client) *Client {
+	return &Client{
+		cli: cli,
+	}
+}
+
+// Get returns the MultitenantPodNetworkConfig identified by the NamespacedName.
+func (c *Client) Get(ctx context.Context, key types.NamespacedName) (*v1alpha1.MultitenantPodNetworkConfig, error) {
+	multitenantPodNetworkConfig := &v1alpha1.MultitenantPodNetworkConfig{}
+	err := c.cli.Get(ctx, key, multitenantPodNetworkConfig)
+	return multitenantPodNetworkConfig, errors.Wrapf(err, "failed to get mpnc %v", key)
+}
+
+// PatchSpec performs a server-side patch of the passed MultitenantPodNetworkConfigSpec to the MultitenantPodNetworkConfig specified by the NamespacedName.
+func (c *Client) PatchSpec(ctx context.Context, key types.NamespacedName, spec *v1alpha1.MultitenantPodNetworkConfigSpec, fieldManager string) (*v1alpha1.MultitenantPodNetworkConfig, error) {
+	obj := genPatchSkel(key)
+	obj.Spec = *spec
+	if err := c.cli.Patch(ctx, obj, client.Apply, client.ForceOwnership, client.FieldOwner(fieldManager)); err != nil {
+		return nil, errors.Wrap(err, "failed to patch mpnc")
+	}
+	return obj, nil
+}
+
+// UpdateSpec does a fetch, deepcopy, and update of the MultitenantPodNetworkConfig with the passed spec.
+// Deprecated: UpdateSpec is deprecated and usage should migrate to PatchSpec.
+func (c *Client) UpdateSpec(ctx context.Context, key types.NamespacedName, spec *v1alpha1.MultitenantPodNetworkConfigSpec) (*v1alpha1.MultitenantPodNetworkConfig, error) {
+	mpnc, err := c.Get(ctx, key)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get mpnc")
+	}
+	spec.DeepCopyInto(&mpnc.Spec)
+	if err := c.cli.Update(ctx, mpnc); err != nil {
+		return nil, errors.Wrap(err, "failed to update mpnc")
+	}
+	return mpnc, nil
+}
+
+// SetOwnerRef sets the controller of the MultitenantPodNetworkConfig to the given object atomically, using HTTP Patch.
+// Deprecated: SetOwnerRef is deprecated, use the more correctly named SetControllerRef.
```

Comment:
Let's not add a new function that's already deprecated

Reply:
removed